packages/bar/src/BarCanvas.tsx

```diff
@@ -52,6 +60,93 @@ const findBarUnderCursor = <RawDatum,>(
 
 const isNumber = (value: unknown): value is number => typeof value === 'number'
 
+function renderTotalsToCanvas<RawDatum extends BarDatum>(
+    ctx: CanvasRenderingContext2D,
+    bars: ComputedBarDatum<RawDatum>[],
+    xScale: ScaleBand<string> | AnyScale,
+    yScale: ScaleBand<string> | AnyScale,
+    layout: BarCommonProps<RawDatum>['layout'] = defaultProps.layout,
+    groupMode: BarCommonProps<RawDatum>['groupMode'] = defaultProps.groupMode
+) {
+    if (bars.length === 0) return
+
+    const totalsByIndex = new Map<string | number, number>()
+
+    const barWidth = bars[0].width
+    const barHeight = bars[0].height
+    const yOffsetVertically = -10
+    const xOffsetHorizontally = 20
+    const fontSize = 12
+
+    ctx.fillStyle = '#222222'
+    ctx.font = `${fontSize}px sans-serif`
+    ctx.textBaseline = 'middle'
+    ctx.textAlign = 'center'
+
+    if (groupMode === 'stacked') {
+        const totalsPositivesByIndex = new Map<string | number, number>()
+
+        bars.forEach(bar => {
+            const { indexValue, value } = bar.data
+            updateTotalsByIndex(totalsByIndex, indexValue, Number(value))
+            updateTotalsPositivesByIndex(totalsPositivesByIndex, indexValue, Number(value))
+        })
+
+        totalsPositivesByIndex.forEach((totalsPositive, indexValue) => {
+            let xPosition: number
+            let yPosition: number
+
+            if (layout === 'vertical') {
+                xPosition = xScale(indexValue)
+                yPosition = yScale(totalsPositive)
+            } else {
+                xPosition = xScale(totalsPositive)
+                yPosition = yScale(indexValue)
+            }
+
+            ctx.fillText(
+                String(totalsByIndex.get(indexValue)),
+                xPosition + (layout === 'vertical' ? barWidth / 2 : xOffsetHorizontally),
+                yPosition + (layout === 'vertical' ? yOffsetVertically : barHeight / 2)
+            )
+        })
+    }
+
+    if (groupMode === 'grouped') {
+        const greatestValueByIndex = new Map<string | number, number>()
```

Comment:
I think it could help top briefly explain the purpose of:
- `totalsByIndex`
- `totalsPositivesByIndex`
- `greatestValueByIndex`
- `numberOfBarsByIndex`
In comments.